### ISSUE TITLE
SuppressionFilter's name attribute format is OS-dependent.

### ISSUE THREAD

**Issue by pbaranchikov** [checkstyle/checkstyle#2270]
Checkstyle's suppression filter identify to ignore using *files* attribute. Values in the attribute use system-dependent path separator.

Reporting from [stackoverflow] (http://stackoverflow.com/questions/13874582/what-is-the-correct-syntax-to-suppress-specific-checkstyle-rules-on-certain-file) suggest very nasty form of workaround:

```xml
<suppress files="my[\\/]super[\\/]package[\\/]name" checks="ModifierOrderCheck|NeedBracesCheck|MagicNumberCheck"/>    
```

There should be an ability to specify uniformed paths, for example in Ant-like style.

**Comment by oburn** [comment]
I like the idea.

You will need to consider that the import-control `<suppress>` element has the attribute `regex` which is used to flag whether the package or class name is interpreted as a regular expression (the example above is missing this).

Suggest consider adding support for a `glob` attribute to support the behaviour your want.

Specifying both `regex` and `glob` as `true` would be an error.

**Comment by romani** [comment]
`glob` - https://en.wikipedia.org/wiki/Glob_%28programming%29

`ant style` - http://ant.apache.org/manual/dirtasks.html#patterns

**Comment by romani** [comment]
@pbaranchikov , @oburn , do you know good implementation of ant and glob matchers that will not blow our dependencies and compliant by License ?

**Comment by MEZk** [comment]
@romani 

Why don't we try default Java NIO PathMatcher:
http://www.concretepage.com/java/jdk7/example-pathmatcher-java-nio
http://javapapers.com/java/glob-with-java-nio/
http://java.boot.by/ocpjp7-upgrade/ch06s05.html
http://www.selikoff.net/2014/01/05/ant-vs-java-using-to-match-directories/

or Apache WildcardFileFilter:
https://commons.apache.org/proper/commons-io/javadocs/api-1.4/org/apache/commons/io/filefilter/WildcardFileFilter.html
http://stackoverflow.com/questions/24433908/apache-commons-i-o-how-to-filter-files-by-wildcard
